src/devices/Media/SoundDevice/Devices/UnixSoundDevice.cs

```diff
@@ -177,114 +207,156 @@ public override void Record(uint recordTimeSeconds, Stream outputStream)
                 SubChunk2 = subChunk2
             };
 
-            WriteWavHeader(outputStream, header);
+            return header;
+        }
 
-            try
+        public override void StartRecording(string outputFilePath)
+        {
+            Record(outputFilePath);
+        }
+
+        public override void StopRecording()
+        {
+            _record = false;
```

Comment:
this would probably be better with Task and CancellationToken

Reply:
What would be the benefit here? All sound is anyway single thread, you can't open 2 streams with Alsa.